src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -160,20 +165,39 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             }
         }
 
-        // TODO would be more efficient to allow API to consolidate writeLog with exitStatus (save an RPC call)
-        @Override public Integer exitStatus(FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-            FilePath status = getResultFile(workspace);
-            if (status.exists()) {
-                try {
-                    return Integer.parseInt(status.readToString().trim());
-                } catch (NumberFormatException x) {
-                    throw new IOException("corrupted content in " + status + ": " + x, x);
+        /** Avoids excess round-tripping when reading status file. */
+        static class StatusCheck extends MasterToSlaveFileCallable<Integer> {
+            @Override
+            @CheckForNull
+            public Integer invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
+                if (f.exists() && f.length() > 0) {
+                    try {
+                        String fileString = Files.readFirstLine(f, Charset.defaultCharset());
```

Comment:
Minor note: use of `defaultCharset` could theoretically cause issues on z/OS. Since we are expecting this file to be ASCII (it should in fact just contain `[0-9]+`), it is safer and clearer to use `StandardCharsets.ASCII`.

Reply:
A really critical point - it looks like this is already broken, per https://github.com/jenkinsci/durable-task-plugin/pull/28 -- I think we'd need to address it in more comprehensive work to get it to play nicely in any case.